ethereum/contracts/relayer/deliveryProvider/DeliveryProvider.sol

```diff
@@ -28,16 +28,19 @@ contract DeliveryProvider is DeliveryProviderGovernance, IDeliveryProvider {
         view
         returns (LocalNative nativePriceQuote, GasPrice targetChainRefundPerUnitGasUnused)
     {
-        targetChainRefundPerUnitGasUnused = gasPrice(targetChain);
-        Wei costOfProvidingFullGasLimit = gasLimit.toWei(targetChainRefundPerUnitGasUnused);
-        Wei transactionFee =
-            quoteDeliveryOverhead(targetChain) + gasLimit.toWei(quoteGasPrice(targetChain));
-        Wei receiverValueCost = quoteAssetCost(targetChain, receiverValue);
-        nativePriceQuote = (
-            transactionFee.max(costOfProvidingFullGasLimit) + receiverValueCost
-        ).asLocalNative();
+        (uint16 buffer, uint16 denominator) = assetConversionBuffer(targetChain);
+        targetChainRefundPerUnitGasUnused = GasPrice.wrap(gasPrice(targetChain).unwrap() * (denominator) / (uint256(denominator) + buffer));
+        
+        TargetNative targetCostIfNoGasLimitUsed = gasLimit.toWei(targetChainRefundPerUnitGasUnused).asTargetNative();
```

Comment:
This calc does not appear to do what the var name suggests. As written it is "how much target chain currency would be refunded if no gas was used", not the cost to the user if no gas was used. 
The name implies the fixed overhead is included, but that is added later in the fn. 

Perhaps, max refund is more descriptive?

Reply:
fixed